### PR TITLE
feat(icon): new icons package

### PR DESCRIPTION
**Pull request by Jerd0**
## Type of change (check all applicable, delete options that are not relevant)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Refactoring (old API is revised and supported)
- [X] Documentation update

### CODE REVIEW

**Review by leonid — CHANGES_REQUESTED** [review]
- svg not exported from /svg folder
- svg are placed in groups, lets use th same approach as in vue directory
- svg export in package.json is wrong
- icons readme is too sparse 
- figma fetcher not providing svgo features
- ui packages dist chunk folders, why we need to change it?
- why we have to remove ui sideeffects on css files? they must be treated as separate chunks in dist and imported in vue chunks

**Comment by Jerd0** [comment]
> * ui packages dist chunk folders, why we need to change it?

In my understanding, everything related to the component should be next to the component and there is no need to move it to a separate chunk, it's strange that our script decides which files to use and which not, if the build can do it

**Comment by leonid** [comment]
> > * ui packages dist chunk folders, why we need to change it?
> 
> In my understanding, everything related to the component should be next to the component and there is no need to move it to a separate chunk, it's strange that our script decides which files to use and which not, if the build can do it

Yes, its ok, i checked it, now all chunks are in the same folder including css styles

**Comment by Jerd0** [comment]
> * why we have to remove ui sideeffects on css files? they must be treated as separate chunks in dist and imported in vue chunks

Adding such a sideEffect to the project configuration serves several important purposes:

Preventing CSS removal:
The main goal is to prevent optimization tools, such as tree shaking in Webpack, from removing CSS files. This is particularly important when CSS is imported globally or through mechanisms that the bundler might not recognize as direct usage.
Preserving global styles:
Many CSS files contain global styles that apply to the entire application. Even if they're not directly imported in JS files, they may be critical to the application's appearance.
Compatibility with CSS-in-JS solutions:
Some CSS-in-JS libraries may dynamically apply styles that are difficult to track through static analysis. This setting ensures that all necessary styles are included.
Working with CSS modules:
When using CSS modules, imports may look non-standard to build tools. This configuration ensures their preservation.
Simplifying configuration:
Instead of specifying each CSS file separately, this setting allows processing all CSS files in the project with a single rule.
Safety during refactoring:
If you remove a CSS file import during development but forget to delete the file itself, this setting will prevent unexpected disappearance of styles from the build.



do we really need one of this cases?

**Comment by Jerd0** [comment]
> * figma fetcher not providing svgo features

do we have knowledge of how we want to flip this svgo config?

**Comment by leonid** [comment]
> > * why we have to remove ui sideeffects on css files? they must be treated as separate chunks in dist and imported in vue chunks
> 
> Adding such a sideEffect to the project configuration serves several important purposes:
> 
> 1. Preventing CSS removal:
>    The main goal is to prevent optimization tools, such as tree shaking in Webpack, from removing CSS files. This is particularly important when CSS is imported globally or through mechanisms that the bundler might not recognize as direct usage.
> 2. Preserving global styles:
>    Many CSS files contain global styles that apply to the entire application. Even if they're not directly imported in JS files, they may be critical to the application's appearance.
> 3. Compatibility with CSS-in-JS solutions:
>    Some CSS-in-JS libraries may dynamically apply styles that are difficult to track through static analysis. This setting ensures that all necessary styles are included.
> 4. Working with CSS modules:
>    When using CSS modules, imports may look non-standard to build tools. This configuration ensures their preservation.
> 5. Simplifying configuration:
>    Instead of specifying each CSS file separately, this setting allows processing all CSS files in the project with a single rule.
> 6. Safety during refactoring:
>    If you remove a CSS file import during development but forget to delete the file itself, this setting will prevent unexpected disappearance of styles from the build.
> 
> do we really need one of this cases?

No, as i already said its already creates proper dist with components folders

**Comment by Jerd0** [comment]
> * figma fetcher not providing svgo features

Do we know how we want to pass this svgo config?

**Comment by leonid** [comment]
> > * figma fetcher not providing svgo features
> 
> do we have knowledge of how we want to pass this svgo config?

Same as before:

- removeViewBox: false
- remove dimensions
- add fill current color attribute
- remove or reduce groups
- prettify svg source
- merge and simplify paths

File: packages/icons/public/workstation.d.ts [hunk 1]
```diff
@@ -1,9 +1,9 @@
-export { default as IconWorkstationNavDark24 } from '../vue/IconWorkstationNavDark24.vue';
 export { default as IconWorkstationNav24 } from '../vue/IconWorkstationNav24.vue';
+export { default as IconWorkstationNavDark24 } from '../vue/IconWorkstationNavDark24.vue';
```
Comment (leonid) [inline packages/icons/public/workstation.d.ts:2]: Wddd

**Pull request merged**